### ISSUE TITLE
Embeddings for the same input does not match

### ISSUE THREAD

**Issue by Ruben-Contesti** [JohnSnowLabs/spark-nlp#12898]
Embeddings for the same input does not match

## Description
I'm running BERT for sentence embedding [1] to a column with words like "Clothing". I'm noticing that for the same word, different embeddings are generated. Sometimes is the 10th of decimal points that is different, but others is the first or second decimal point.

[1] https://nlp.johnsnowlabs.com/2020/08/25/sent_small_bert_L8_512.html

## Expected Behavior
Embeddings generated for the same string should always be the same

## Current Behavior
Embeddings generated for the same string are not always the same

## Steps to Reproduce
````
import sparknlp
from nlp.featurization import Featurizer
from pyspark.sql.functions import regexp_replace, col, trim

from sparknlp.annotator import (
    SentenceDetector,
    BertSentenceEmbeddings,
)
from pyspark.ml import Pipeline
from sparknlp.base import DocumentAssembler, EmbeddingsFinisher

spark=sparknlp.start()

path_model = "/models/bert_small_sentence_embeddings"

input_col = "text"
output_col = "features"

document_assembler = (
        DocumentAssembler().setInputCol(input_col).setOutputCol("document")
    )

sentence_detector = SentenceDetector().setInputCols(["document"]).setOutputCol("sentence")

embeddings = BertSentenceEmbeddings.load(path_model) \
.setInputCols("sentence") \
.setOutputCol("sentence_embeddings")

embeddingsFinisher = (
    EmbeddingsFinisher()
    .setInputCols("sentence_embeddings")
    .setOutputCols(output_col)
    .setOutputAsVector(True)
)

df = spark.createDataFrame([['Clothing'], ["Clothing"], ["Clothing"], ["Clothing"],["Clothing"]], ["text"])


nlp_pipeline = Pipeline(stages=[document_assembler, sentence_detector, embeddings, embeddingsFinisher])
pipeline_model = nlp_pipeline.fit(df)
result = pipeline_model.transform(df)
result = result.select("features")

result.show()
````

+--------------------+                                                          
|            features|
+--------------------+
|[[-1.694393515586...|
|[[-1.694393515586...|
|[[-1.694393157958...|
|[[-1.694392800331...|
|[[-1.694392800331...|
|[[-1.694393515586...|
|[[-1.694393515586...|
|[[-1.694393157958...|
|[[-1.694392800331...|
|[[-1.694392800331...|
+--------------------+

If instead, I use the pre-trained version (downloading from the Internet), The results are different and still inconsistent:

+--------------------+                                                          
|            features|
+--------------------+
|[[0.2541341483592...|
|[[0.2541341483592...|
|[[0.2541342973709...|
|[[0.2541346251964...|
|[[0.2541346251964...|
|[[0.2541341483592...|
|[[0.2541341483592...|
|[[0.2541342973709...|
|[[0.2541346251964...|
|[[0.2541346251964...|
+--------------------+

## Your Environment

* Spark NLP version `sparknlp.version()`: '4.0.2'
* Apache NLP version `spark.version`:
* Java version `java -version`:
* Setup and installation (Pypi, Conda, Maven, etc.):
* Operating System and version: Linux
* Link to your project (if any):

**Comment by maziyarpanahi** [comment]
Hi @Ruben-Contesti 

Thanks for reporting this. Just have one questions before I dig in:
- When you say "If instead, I use the pre-trained version", what is `/models/bert_small_sentence_embeddings`? (isn't it the same model online, or does it download a newer version?)

A quick note, `sent_small_bert_L8_512` is coming from TF Hub and we just use the `pooling` layer instead of `last_hidden_layer`. Once I got an answer to what is the saved model being `loaded` I can then see if there is something wrong with the model itself coming from TF Hub or something else.

**Comment by Ruben-Contesti** [comment]
Hi @maziyarpanahi. You are right. It is supposed to be the same model. But the outputs differ. The only thing I did was to download the model and run it locally. In both cases the model is not capable of producing the same output for the same input.

**Comment by Ruben-Contesti** [comment]
I also tried the Universal Sentence Encoder, and again embeddings are not the same. https://nlp.johnsnowlabs.com/2020/04/17/tfhub_use.html. The larger the number of rows the more dissimilar is the output

````
+-----------+--------------------+                                              
|       text|            features|
+-----------+--------------------+
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983061015...| << Different
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|   Clothing|[[-0.078983046114...|
|      Cloth|[[-0.012505427002...|
|    Apparel|[[-0.043411109596...|
+-----------+--------------------+

````

**Comment by maziyarpanahi** [comment]
I think we can see if we should set a `seed` or that much error in precision points in  Float conversion is unavoidable. (most definitely not an issue in actual prediction where we have trained very high-accurate models by using these embeddings).
Will dig into this more

**Comment by Ruben-Contesti** [comment]
In particular, for the dataset I'm using, the precision is modified to the first and second decimal points:

````
+-----------------------+-------------------------------+-----+
|product_category_PREPRO|product_category_PREPRO_FEATURE|count|
+-----------------------+-------------------------------+-----+
|               Clothing|           [[-1.734645128250...|    9|
|               Clothing|           [[-1.565135002136...|    5|
|               Clothing|           [[-1.472201228141...|    5|
|               Clothing|           [[-1.728367090225...|    4|
|               Clothing|           [[-1.642018675804...|    2|
|               Clothing|           [[-1.642018675804...|    1|
|               Clothing|           [[-1.271631836891...|    1|
+-----------------------+-------------------------------+-----+

````

**Comment by maziyarpanahi** [comment]
We will use the same code to reproduce and see first if setting `seed` helps. If so, we will introduce `setSeed()` in all the embeddings. 
I will come back to you

**Comment by maziyarpanahi** [comment]
- I have managed to reproduce this small discrepancy in Python
- This doesn't happen in Scala  
<img width="978" alt="image" src="https://user-images.githubusercontent.com/5762953/194774216-9714b089-df1f-481a-a250-44d7303fa579.png">

That said, I have tried to use these embeddings to train the same datasets with both Python and Scala, and the results were identical. This shows this tiny discrepancy won't impact the downstream tasks. However, I am going to dig deeper to see why going from Scala to Python we see this and whether or not it's possible to avoid this.
